spring-cloud-deployer-resource-support/src/main/java/org/springframework/cloud/deployer/resource/support/HttpResource.java

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright 2017 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.cloud.deployer.resource.support;
+
+import java.io.File;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.net.MalformedURLException;
+import java.net.URI;
+import java.nio.file.Files;
+
+import org.springframework.core.io.AbstractResource;
+import org.springframework.core.io.Resource;
+import org.springframework.core.io.UrlResource;
+import org.springframework.util.Assert;
+import org.springframework.util.FileCopyUtils;
+
+/**
+ * A {@link Resource} implementation for resolving HTTP resource.
+ *
+ * @author Ilayaperumal Gopinathan
+ */
+public class HttpResource extends UrlResource {
+
+	private final String location;
+
+	public HttpResource(String location) throws MalformedURLException {
+		super(location);
+		this.location = location;
+	}
+
+	@Override
+	public String getDescription() {
+		return "Http Resource [" + location + "]";
+	}
+
+	@Override
+	public File getFile() throws IOException {
```

Comment:
There should be some tests for this method.
I don't see why we need SHA anymore, it makes things more opaque.  Similarly, we could group all these files under a directory "spring-cloud-deployer" instead of having them as top level in the temp dir.  This method would also benefit from being synchronized.

Reply:
Also, was reading a bit more into temp files...

https://www.pongasoft.com/blog/yan/java/2011/05/17/file-dot-deleteOnExit-is-evil/

is a quick read.  We should prob remove the deleteOnExit.    We can do the disk mgmt via https://github.com/spring-cloud/spring-cloud-dataflow-server-cloudfoundry/blob/master/spring-cloud-dataflow-server-cloudfoundry-autoconfig/src/main/java/org/springframework/cloud/dataflow/server/cloudfoundry/resource/LRUCleaningResourceLoader.java 

perhaps  `LRUCleaningResourceLoader` needs to move into spring-cloud-deployer...

Reply:
Also, just generally looking over `Resource` implementation in core spring.  We probably need to override some other methods as well, e.g. `getOutputStream`, `isWritable`, `createRelative` so that is is a "true" Resource implementation vs. one that just happens to work for the deployer use case ATM.    I think that we may need to keep the file that is downloaded 'on demand' in a private field, so that other methods can reference it.  For example, getOutputStream should not have to download the file again.

Reply:
It looks like `getInputStream` always invokes the download of Http resource which I believe is sufficient along with having a new temp file every time getFile is invoked. Do you see any issue with that?

Reply:
Pushed a test that shows the temp files are distinct. Also, it looks like `deleteOnExit` is unavoidable at least for now as that seems to be the only possible way to delete those temp files. Given, we don't explicitly invoke delete on these files, possibility leakage isn't there I believe.